engine/src/main/java/org/hibernate/search/engine/search/predicate/dsl/QueryStringPredicateOptionsStep.java

```diff
@@ -0,0 +1,202 @@
+	//	 */
+	//	S fuzziness(int fuzziness);
+
+
+	//	/**
+	//	 * The maximum number of terms to which the query can expand.
+	//	 * Fuzzy queries “expand to” a number of matching terms that are within the distance specified in fuzziness.
+	//	 * Then OpenSearch tries to match those terms. Default is 50.
+	//	 * @param fuzzyMaxExpansions
+	//	 * @return {@code this}, for method chaining.
+	//	 */
+	//	S fuzzyMaxExpansions(int fuzzyMaxExpansions);
+
+	//	/**
+	//	 * Setting fuzzy_transpositions to true (default) adds swaps of adjacent characters to the insert, delete,
+	//	 * and substitute operations of the fuzziness option.
+	//	 * For example, the distance between wind and wnid is 1 if fuzzy_transpositions is true (swap “n” and “i”) and 2
+	//	 * if it is false (delete “n”, insert “n”). If fuzzy_transpositions is false,
+	//	 * rewind and wnid have the same distance (2) from wind,
+	//	 * despite the more human-centric opinion that wnid is an obvious typo.
+	//	 * The default is a good choice for most use cases.
+	//	 *
+	//	 * @param fuzzyTranspositions
+	//	 * @return {@code this}, for method chaining.
+	//	 */
+	//	S fuzzyTranspositions(boolean fuzzyTranspositions);
+
+	//	/**
+	//	 * Setting lenient to true ignores data type mismatches between the query and the document field.
+	//	 * For example, a query string of "8.2" could match a field of type float. Default is false.
+	//	 *
+	//	 * @param lenient
+	//	 * @return {@code this}, for method chaining.
+	//	 */
+	//	S lenient(boolean lenient);
+
+	//	/**
+	//	 *
+	//	 * @param maxDeterminizedStates
+	//	 * @return {@code this}, for method chaining.
+	//	 */
+	//	S maxDeterminizedStates(int maxDeterminizedStates);
+
+	/**
+	 *
+	 * @param matchingClausesNumber
+	 * @return {@code this}, for method chaining.
+	 */
+	default S minimumShouldMatchNumber(int matchingClausesNumber) {
+		return minimumShouldMatch()
+				.ifMoreThan( 0 ).thenRequireNumber( matchingClausesNumber )
+				.end();
+	}
+
+	/**
+	 *
+	 * @param matchingClausesPercent
+	 * @return {@code this}, for method chaining.
+	 */
+	default S minimumShouldMatchPercent(int matchingClausesPercent) {
+		return minimumShouldMatch()
+				.ifMoreThan( 0 ).thenRequirePercent( matchingClausesPercent )
+				.end();
+	}
+
+	MinimumShouldMatchConditionStep<? extends S> minimumShouldMatch();
+
+	/**
+	 *
+	 * @param constraintContributor
+	 * @return {@code this}, for method chaining.
+	 */
+	S minimumShouldMatch(
+			Consumer<? super MinimumShouldMatchConditionStep<?>> constraintContributor);
+
+	/**
+	 * The maximum number of words that are allowed between the matched words.
+	 * If phrase_slop is 2, a maximum of two words is allowed between matched words in a phrase.
+	 * Transposed words have a slop of 2.
+	 * Default is 0 (an exact phrase match where matched words must be next to each other).
+	 *
+	 * @param phraseSlop
+	 * @return {@code this}, for method chaining.
+	 */
+	S phraseSlop(Integer phraseSlop);
```

Comment:
There's a phrase predicate in Hibernate Search, please make sure the naming/javadoc is consistent with that predicate (I didn't check).